scraper/src/github-scraper/discussion.ts

```diff
@@ -37,54 +36,111 @@ query($org: String!, $cursor: String) {
                   }
                 }
                 createdAt
+                updatedAt
               }
             }
           }
         }
       }
     }
   }
-}
-`;
+}`;
 
-async function fetchDiscussionsForOrg(org: string, cursor = null) {
-  const variables = { org, cursor };
-  const response = await octokit.graphql.paginate(query, variables);
+async function fetchGitHubDiscussions(
+  org: string,
+  endDate: Date,
+  startDate: Date,
+  cursor = null,
+) {
+  const variables = {
+    org,
+    cursor,
+  };
+  for await (const response of octokit.graphql.paginate.iterator(
+    query,
+    variables,
+  )) {
+    const repositories = await response.organization.repositories.edges;
+    type repo = (typeof repositories)[0];
+    for (const repo of repositories) {
+      const discussions = await repo.node.discussions.edges.map(
+        (discussion: repo) => ({
+          repoName: repo.node.name,
+          discussion: discussion.node,
+        }),
+      );
+      const discussionsWithinDateRange = await discussions.find((d: repo) => {
+        const createdAt = new Date(d.discussion.createdAt);
+        const updatedAt = new Date(d.discussion.updatedAt);
 
-  type Edge = typeof response.organization.repositories.edges;
-  const discussions = response.organization.repositories.edges.map(
-    (edge: Edge) => edge.node.discussions.edges,
-  );
+        return (
+          createdAt >= new Date(startDate) || updatedAt >= new Date(startDate)
+        );
+      });
+      if (discussionsWithinDateRange) {
+        return discussions;
+      }
```

Comment:
Are you sure this fetches all discussions? I don't see any logic that accumulates valid discussions that passes the filter rules and return them all. This just returns as soon as the first discussions within range is valid right? What about the rest of the pages?

Reply:
Ohh i did not noticed that 

So the logic must be ``createdAt < new Date(startDate) || updatedAt < new Date(startDate)` If it true than we reach the page where the discussions are older than the start date because we are fetching based last updated and in desc order

Correct me if i am wrong

Reply:
It's not about the filter logic. It's about when/where/how you are using the return statement.

Does it return all the discussions within the date range or just the valid discussions from the first page alone?

Reply:
Got it,
Bro so need to create a new variable and whenever it is within range than append with it
and if the date range is hit than that new variable will be returned so that way we have correct array of discussion and good use of iterator

Reply:
Here is the updated code 
![image](https://github.com/user-attachments/assets/b635ae37-9b20-4c6a-9136-283702ab2bfa)
I guess this will solve all the above problems all type related issues and naming conventions issues and logical issues

Reply:
The issue still remains same. THe logic for createdAt startDate updatedAt endDate is fine. But the issue is about: "Does it return all the discussions within the date range or just the valid discussions from the **first page alone**?"

Reply:
As per the above logic this is the debug view how it works please give it try once 
![image](https://github.com/user-attachments/assets/e4c5f52f-aae8-4668-a854-33eec2c28639)